default/scripting/buildings/IMPERIAL_GARDENS.focs.txt

```diff
@@ -0,0 +1,28 @@
+BuildingType
+    name = "BLD_IMPERIAL_GARDENS"
+    description = "BLD_IMPERIAL_GARDENS_DESC"
+    buildcost = 150 * [[BUILDING_COST_MULTIPLIER]]
+    buildtime = 5
+    location = And [
+        Planet
+        Contains Building name = "BLD_IMPERIAL_PALACE"
+        Not Contains Building name = "BLD_IMPERIAL_GARDENS"
+        OwnedBy empire = Source.Owner
+        TargetPopulation low = 1
+    ]
+    EnqueueLocation = [[ENQUEUE_BUILD_ONE_PER_PLANET]]
+    effectsgroups = [
+        EffectsGroup
+            scope = And [
+                Object id = Source.PlanetID
+                Planet
+                NOT Population high = 0
+                OwnedBy empire = Source.Owner
+            ]
+            activation = Source
+            effects = SetTargetPopulation value = Value * 1.5
```

Comment:
This is looking like a really large effect to me.  Particularly since it will scale up as technology advances, it seems significant enough that folks would feel pressured to move their Palace around so that they could build these everywhere.  I haven't play-tested this PR yet, so I can't speak to the 'slowing down' from the other current changes that you cite as the reason for adding this building and the workshop, but if something along these lines is truly needed I think it would likely be best as an addition rather than multiple, either a constant number or some multiple of Habitable Size.  For other population effect references, take a look at https://www.freeorion.org/index.php/EffectsGroup_Priority_Standard_Values

Reply:
to at least help reduce the 'build everywhere' problem you might make it require both the palace and archives be present

Reply:
Whoa, i didn't think of "moving the palace around", that is not intended, and that would indeed be unacceptable micromanagement.  So i have to change this and either bind it to the Cultural Archives instead (which you cannot move around, right?) or put back the Activation rule such that it only takes effect when the palace is present, like it is done for the BLD_AUTO_HISTORY_ANALYSER.  The idea of making it multiplicative rather  than additive is to not de-balance species with population boni/mali or smaller or larger starting planets.  Also, you are right that the intention is *not* to have the effect scale with technological progress, so it should probably be TARGET_POPULATION_SCALING_PRIORITY.  I will reconsider and probably push a patch to fix that.  Thanks for the insight!

Reply:
Done in commit e89c6e76 (allowing only one per empire, setting the priority, making it additive, and weakening it a bit).